x-pack/plugins/ml/index.js

```diff
@@ -81,9 +82,14 @@ export const ml = (kibana) => {
         // to re-compute the license check results for this plugin
         xpackMainPlugin.info.feature(thisPlugin.id).registerLicenseCheckResultsGenerator(checkLicense);
 
+        // Add links to the Kibana sample data sets if ml is enabled
+        // and there is a full license (trial or platinum).
         const isEnabled = xpackMainPlugin.info.feature(thisPlugin.id).isEnabled();
         if (isEnabled === true) {
-          addLinksToSampleDatasets(server);
+          const licenseCheckResults = xpackMainPlugin.info.feature(thisPlugin.id).getLicenseCheckResults();
```

Comment:
`xpackMainPlugin.info.feature(thisPlugin.id)` is called 3 times in this function. 
it might be worth creating a `const feature` variable to store the result and using that instead.